nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -348,10 +360,11 @@ def setup_test_data(self, cfg):
             consumed_samples = 0
             self._test_dl = self.build_pretraining_data_loader(self._test_ds, consumed_samples)
 
-    def compute_consumed_samples(self, global_step):
+    def compute_consumed_samples(self, global_relative_step=0):
         app_state = AppState()
         consumed_samples = (
-            global_step
+            self.init_consumed_samples
+            + global_relative_step
             * app_state.data_parallel_size
             * self._cfg.micro_batch_size
             * self.trainer.accumulate_grad_batches
```

Comment:
Why not using here `get_num_microbatches()` like GPT?

Reply:
Need to update to pipeline parallel for that.